programs/marginfi/src/instructions/marginfi_account/initialize.rs

```diff
@@ -49,3 +52,75 @@ pub struct MarginfiAccountInitialize<'info> {
 
     pub system_program: Program<'info, System>,
 }
+
+pub fn initialize_account_pda(
+    ctx: Context<MarginfiAccountInitializePda>,
+    account_index: u16,
+    third_party_id: Option<u16>,
+) -> MarginfiResult {
+    let MarginfiAccountInitializePda {
+        authority,
+        marginfi_group,
+        marginfi_account: marginfi_account_loader,
+        ..
+    } = ctx.accounts;
+
+    if let Some(id) = third_party_id {
+        if !is_allowed_cpi_for_third_party_id(&ctx.accounts.instructions_sysvar, id)? {
+            return err!(MarginfiError::Unauthorized);
+        }
+    }
+
+    let mut marginfi_account = marginfi_account_loader.load_init()?;
+
+    marginfi_account.initialize(marginfi_group.key(), authority.key());
+    marginfi_account.account_index = account_index;
+    marginfi_account.third_party_index = third_party_id.unwrap_or(0);
+    marginfi_account.bump = ctx.bumps.marginfi_account;
+
+    emit!(MarginfiAccountCreateEvent {
+        header: AccountEventHeader {
+            signer: Some(authority.key()),
+            marginfi_account: marginfi_account_loader.key(),
+            marginfi_account_authority: marginfi_account.authority,
+            marginfi_group: marginfi_account.group,
+        }
+    });
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+#[instruction(account_index: u16, third_party_id: Option<u16>)]
+pub struct MarginfiAccountInitializePda<'info> {
+    pub marginfi_group: AccountLoader<'info, MarginfiGroup>,
+
+    #[account(
+        init,
+        payer = fee_payer,
+        space = 8 + std::mem::size_of::<MarginfiAccount>(),
+        seeds = [
+            MARGINFI_ACCOUNT_SEED.as_bytes(),
+            marginfi_group.key().as_ref(),
+            authority.key().as_ref(),
+            &account_index.to_le_bytes(),
+            &third_party_id.unwrap_or(0).to_le_bytes(),
+        ],
+        bump
+    )]
+    pub marginfi_account: AccountLoader<'info, MarginfiAccount>,
+
+    /// CHECK: Authority is only used for PDA seed derivation, no signing required
+    pub authority: UncheckedAccount<'info>,
```

Comment:
nit - In the old instruction this used to be required as a signer and all other platforms require the account whose authority it is to be a signer

Reply:
We have account transfers now, so do we care about that? You can always immediately transfer the account to a new authority (though I guess in that instance you at least pay us $0.50 to do it). I like supporting the use-case where the authority does not yet exist (tho transfer does the same thing).

On one hand you could spam-create accounts for somebody relatively cheaply. On the other hand since they are PDAs now it's really trivial to capture them all and then close them for rent so it's a pretty silly troll.

IDK @IliaZyrin you wanna weigh in?